### PR TITLE
Commit dialog honors value of `status.showUntrackedFiles'  git config.

### PR DESCRIPTION
**Pull request by mlutescu**
Hello, 

This is first time when I use GitHub to contribute to somebody's else project, also I have practically 0 experience with C#; so please forgive me if I do something patently wrong.  My small contribution tries to add feature described in https://github.com/gitextensions/gitextensions/issues/1872.

### CODE REVIEW

File: GitUI/CommandsDialogs/FormCommit.cs [hunk 1]
```diff
@@ -229,6 +229,7 @@ private void FormCommit_Load(object sender, EventArgs e)
             ResetUnStaged.Visible = AppSettings.ShowResetUnstagedChanges;
             CommitAndPush.Visible = AppSettings.ShowCommitAndPush;
             AdjustCommitButtonPanelHeight();
+	    showUntrackedFilesToolStripMenuItem.Checked = !Module.RunGitCmd("config status.showUntrackedFiles").Trim().Equals("no");
```
Comment (KindDragon) [inline GitUI/CommandsDialogs/FormCommit.cs]: You should use Module.EffectiveConfigFile instead and we use 4 space indentation.

**Comment by mlutescu** [comment]
Hello Arkady,

Thanks for reviewing my commit and for the advise.  I implemented your suggestion and I use Module.EffectiveConfigFile.  Also I corrected the spacing problem.

It looks like I get into a bug for the mechanism that Module.EffectiveConfigFile uses to read the settings: it does not cope well with multiple occurrences of the same section, a situation which someone may get into if <em>.git/config</em> is manipulated via <code>git config</code> as I described in my second commit comment.  Please advise.

**Comment by mlutescu** [comment]
Should I close this pull request an make a new one?

File: GitUI/CommandsDialogs/FormCommit.cs [hunk 2]
```diff
@@ -229,7 +229,7 @@ private void FormCommit_Load(object sender, EventArgs e)
             ResetUnStaged.Visible = AppSettings.ShowResetUnstagedChanges;
             CommitAndPush.Visible = AppSettings.ShowCommitAndPush;
             AdjustCommitButtonPanelHeight();
-	    showUntrackedFilesToolStripMenuItem.Checked = !Module.RunGitCmd("config status.showUntrackedFiles").Trim().Equals("no");
+            showUntrackedFilesToolStripMenuItem.Checked = !Module.EffectiveConfigFile.GetValue("status.showUntrackedFiles").Equals("no");
```
Comment (KindDragon) [inline GitUI/CommandsDialogs/FormCommit.cs]: Better use string.Equals(val, "astringvalue", StringComparison.OrdinalIgnoreCase) to ignore case

**Comment by KindDragon** [comment]
> Should I close this pull request an make a new one?

Just squash commits and force push to your branch

**Comment by KindDragon** [comment]
> It looks like I get into a bug for the mechanism that Module.EffectiveConfigFile uses to read the settings: it does not cope well with multiple occurrences of the same section, a situation which someone may get into if .git/config is manipulated via git config as I described in my second commit comment. Please advise.

Please create separate bug about that if you can't fix it currently

**Pull request merged**